src/main/scala/chisel3/util/LTL.scala

```diff
@@ -0,0 +1,392 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.ltl
+
+import chisel3._
+import chisel3.util.circt._
+import chisel3.experimental.hierarchy.{Instance, Instantiate}
+
+/** An opaque sequence returned by an intrinsic.
+  *
+  * Due to the lack of opaque user-defined types in FIRRTL, the Linear Temporal
+  * Logic (LTL) intrinsics operate on `Bool` values instead of a more desirable
+  * `Sequence` type. To prevent abuse, the Chisel API here wraps these `Bool`s
+  * in an opaque type over which we can provide a safe interface that cannot be
+  * used to construct invalid IR.
+  */
+private case class OpaqueSequence(private[ltl] val inner: Bool) extends Sequence
+
+/** An opaque property returned by an intrinsic.
+  *
+  * See `OpaqueSequence` for details.
+  */
+private case class OpaqueProperty(private[ltl] val inner: Bool) extends Property
+
+/** A single item that may be used in the `Sequence(...)` convenience
+  * constructor for sequences. These atoms may either be `Sequence`s themselves,
+  * like `a` or `a and b`, or a `DelayAtom`, like `###`. Together they enable a
+  * shorthand notation for sequences like:
+  *
+  * `Sequence(a, ###, b, ###(2), c, ###(3, 9), d, ###(4, None))`.
```

Comment:
This syntax is likely to be confusing to anyone the first time they see it.

I thought that `###` was intended to be an infix operator or something when I saw this at first glance. But if it's being used like in this example, what compelling reason is there to use a symbol instead of just calling it `Delay`.

Reply:
The syntax tries to stay close to SVA sequences, where it would read as:
```
a ## b ##2 c ##[3:9] d ###[4:$] e
```
(`e` is missing in the comment)

So `###` actually is an infix operator. It's just that `b ###(2) c` doesn't work in scala. So the closest I could get to providing infix notation is to fake it by accepting sequences and the delay infix operators as individual atoms in the `Sequence(...)` argument list.

Reply:
Another option is to make the pure Scala API just use a named thing, like `Delay`, so it could instead be:
```scala
Sequence(a, Delay(), b, Delay(2), c, Delay(3, 9), d, Delay(4, None))
```
And then use a custom String interpolator to provide a compile-time checked SVA DSL:
```
sva"$a ## $b ##2 $c ##[3:9] $d ###[4:$] e"
```
It may even be possible to elide the `$`, that I'm unsure of, but I know the above can work and even give pretty Scala compile-time errors when used incorrectly.

Reply:
Yeah I could totally switch this over to a named thing, and then we can add the string interpolator as a follow-up!